pkg/kubeclient/kubelet_client.go

```diff
@@ -166,6 +169,9 @@ func (client *KubeletClient) GetMachineInfo(host string) (*cadvisorapi.MachineIn
 		if entryPresent {
 			glog.V(2).Infof("unable to retrieve machine[%s] machine info: %v. Using cached value", host, err)
 			entry.used = true
+			if entry.machineInfo == nil {
+				return nil, err
```

Comment:
Same here. let's add another log message here to indicate that cache is used ,but machineinfo is nil. Otherwise we can't distinguish the case whether we populate the machineinfo from the cache or not.